cli/glnexus_cli.cc

```diff
@@ -490,6 +490,258 @@ int main_dump(int argc, char *argv[]) {
     return 0;
 }
 
+
+GLnexus::Status load_discovered_alleles_and_ranges(const std::string& name,
+                                                   const std::vector<std::pair<std::string,size_t> > &contigs,
+                                                   vector<GLnexus::range> &ranges,
+                                                   vector<GLnexus::discovered_alleles> &valleles) {
+    GLnexus::Status s;
+    if (name.size() == 0) {
+        return GLnexus::Status::Invalid("The discovered alleles file must be specified");
+    }
+
+    cerr << "Loading discovered alleles "<< endl;
+    YAML::Node yaml = YAML::LoadFile(name);
+    if (!yaml) {
+        return GLnexus::Status::NotFound("bad discovered alleles file", name);
+    }
+    if (!yaml.IsMap()) {
+        return GLnexus::Status::Invalid("not a map at top level");
+    }
+
+    ranges.clear();
+    valleles.clear();
+
+    const auto n_ranges = yaml["ranges"];
+    for (YAML::const_iterator p = n_ranges.begin(); p != n_ranges.end(); ++p) {
+        GLnexus::range rng(-1,-1,-1);
+        s = range_of_yaml(*p, contigs, rng);
+        if (s.bad()) return s;
+        ranges.push_back(rng);
+    }
+    console->info() << "read " << ranges.size() << " ranges";
+
+    const auto n_dsals = yaml["discovered_alleles"];
+    for (YAML::const_iterator p = n_dsals.begin(); p != n_dsals.end(); ++p) {
+        GLnexus::discovered_alleles dsals;
+        s = discovered_alleles_of_yaml(*p, contigs, dsals);
+        if (s.bad()) return s;
+        valleles.push_back(dsals);
+    }
+
+    unsigned discovered_allele_count=0;
+    for (const auto& dsals : valleles) {
+        discovered_allele_count += dsals.size();
+    }
+    console->info() << "read " << discovered_allele_count << " discovered alleles";
```

Comment:
How about a serialization something like
```
- containing_range: {ref: '22', beg: 1000, end: 1100}
  discovered_alleles:
    <yaml serialization of discovered_alleles here>
- containing_range: {ref: '22', beg: 2000, end: 2100}
  discovered_alleles:
    <yaml serialization of discovered_alleles here>
```

This would keep stuff together and avoid any need to worry about are the vectors the same length with corresponding items